src/lib/KeyStore.js

```diff
@@ -102,26 +119,40 @@ class KeyStore {
      * @returns {Promise<void>}
      */
     async put(key, passphrase) {
-        const secret = !passphrase
-            ? key.secret
-            : await Nimiq.CryptoUtils.encryptOtpKdf(new Nimiq.SerialBuffer(key.secret), passphrase);
+        /** @type {Uint8Array} */
+        let secret;
+        if (passphrase) {
+            secret = await key.secret.exportEncrypted(passphrase);
+        } else {
+            const buf = new Nimiq.SerialBuffer(KeyStore.UNENCRYPTED_SECRET_SIZE);
+            const purposeId = key.secret.type === Nimiq.Secret.Type.ENTROPY
```

Comment:
Same as above, but here we are actually using `key.secret.type`.

Reply:
Switched it around and added a comment about why the purposeID is stored.